### PR TITLE
security init commit

### PR DESCRIPTION
**Pull request by TonyH277**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/config/security/SecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        return http
+                .cors(AbstractHttpConfigurer::disable)
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        auth -> auth
+                                .requestMatchers("/auth/**", "/error")
```
Comment (makaroshyna) [inline src/main/java/mate/academy/bookshop/config/security/SecurityConfig.java]: I think we should put "/swagger-ui/**" here too.

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 2]
```diff
@@ -30,27 +31,31 @@ public class BookController {
 
     @Operation(summary = "Get all books",
             description = "Get certain page of books with custom size")
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (makaroshyna) [inline src/main/java/mate/academy/bookshop/controller/BookController.java]: I think Admin should be able to get all books.

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 3]
```diff
@@ -30,27 +31,31 @@ public class BookController {
...
     @GetMapping
     public List<BookDto> getAll(Pageable pageable) {
         return bookService.findAll(pageable);
     }
 
     @Operation(summary = "Get book by id",
             description = "Get book by id")
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (makaroshyna) [inline src/main/java/mate/academy/bookshop/controller/BookController.java]: And here too, why shouldn't admin have an access to get endpoints?

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 4]
```diff
@@ -30,27 +31,31 @@ public class BookController {
...
     @GetMapping("/{id}")
     public BookDto getById(@PathVariable Long id) {
         return bookService.findBookById(id);
     }
 
     @Operation(summary = "Get books which match searchParameters",
             description = "Get books which match searchParameters")
+    @PreAuthorize("hasRole('ROLE_USER')")
```
Comment (makaroshyna) [inline src/main/java/mate/academy/bookshop/controller/BookController.java]: The same here.

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 5]
```diff
@@ -26,4 +37,41 @@ public class User {
     @Column(nullable = false)
     private String lastName;
     private String shippingAddress;
+    @Column(name = "is_deleted")
+    private boolean isDeleted;
```
Comment (makaroshyna) [inline src/main/java/mate/academy/bookshop/model/User.java:41]:
```suggestion
    private boolean isDeleted = false;
```

File: src/main/java/mate/academy/bookshop/model/User.java [hunk 6]
```diff
@@ -26,4 +37,41 @@ public class User {
...
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles;
+    }
+
+    @Override
+    public String getUsername() {
+        return email;
+    }
+
```
Comment (makaroshyna) [inline src/main/java/mate/academy/bookshop/model/User.java:57]: I think you should also add getPassword() here.

**Review by makaroshyna — CHANGES_REQUESTED** [review]
Good job! Just some minor comments here.

File: src/main/resources/db/changelog/changes/06-create-users-roles-join-table.yaml [hunk 7]
```diff
@@ -0,0 +1,20 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-users-role-join-table
+      author: Anton Haiduk
+      changes:
+        - createTable:
+            tableName: users_roles
```
Comment (vovasalyha) [inline src/main/resources/db/changelog/changes/06-create-users-roles-join-table.yaml:7]: Make user_id + role_id a composite primary key to disallow duplicate entries.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**